### ISSUE TITLE
InvalidOperationException thrown from Cancel

### ISSUE THREAD

**Issue by bgrainger** [mysql-net/MySqlConnector#386]
```
System.InvalidOperationException: Expected state to be (Querying|CancelingQuery) but was Failed.
   at MySql.Data.Serialization.MySqlSession.VerifyState(State state1, State state2)
   at MySql.Data.Serialization.MySqlSession.TryStartCancel(MySqlCommand command)
   at MySql.Data.MySqlClient.MySqlConnection.Cancel(MySqlCommand command)
   at Dapper.SqlMapper.GridReader.Dispose()
   at [Class].<MethodAsync>d__5.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
```

It's possible that the command is timing out (and killing the connection). If so, Cancel should be a no-op.

**Comment by bgrainger** [comment]
Couldn't create a test to reproduce it, but fixed the logic error that allowed this to happen.

**Comment by bgrainger** [comment]
Fixed in [0.31.0](https://github.com/mysql-net/MySqlConnector/releases/tag/0.31.0).
